cmd/kubeadm/app/cmd/join.go

```diff
@@ -262,6 +262,19 @@ func (j *Join) Run(out io.Writer) error {
 		return fmt.Errorf("couldn't save the CA certificate to disk: %v", err)
 	}
 
+	// read the kubeadm log file and write a new one if it doesn't exist
+	data, err := kubeadmutil.ReadKubeadmLogFile()
```

Comment:
can you make the semantics append only and you can eliminate these check in here and place error logging into the utility.  e.g. consider it almost like glog.

Reply:
we can technically have a function in the lines of:
```
AppendToKubeadmLog("kubeadm init")
```

and inside the function there could logging, verification that "kubeadm init" or "join" were not called before and also use of log.Fatal() for errors without returning to the caller. but in that case the naming should be more like `AppendToKubeadmLogVerify()`, because it also does verification.

should it not even return errors to the parent?
if you insist we can go with a single function?

Reply:
It should not error to parent, it's just like glog.